ush/python/pygfs/task/land_analysis.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python3
+
+import os
+import glob
+import gzip
+import tarfile
+from logging import getLogger
+from typing import Dict, List, Any
+
+from pygw.attrdict import AttrDict
+from pygw.file_utils import FileHandler
+from pygw.timetools import add_to_datetime, to_fv3time, to_timedelta
+from pygw.fsutils import rm_p, chdir
+from pygw.timetools import to_fv3time
+from pygw.yaml_file import YAMLFile, parse_yamltmpl, parse_j2yaml, save_as_yaml
+from pygw.logger import logit
+from pygw.executable import Executable
+from pygw.exceptions import WorkflowException
+from pygfs.task.analysis import Analysis
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class LandAnalysis(Analysis):
+    """
+    Class for global land analysis tasks
+    """
+    @logit(logger, name="LandAnalysis")
+    def __init__(self, config):
+        super().__init__(config)
+
+        _res = int(self.config['CASE'][1:])
+        _res_enkf = int(self.config['CASE_ENKF'][1:])
+        _window_begin = add_to_datetime(self.runtime_config.current_cycle, -to_timedelta(f"{self.config['assim_freq']}H") / 2)
+        _fv3jedi_yaml = os.path.join(self.runtime_config.DATA, f"{self.runtime_config.CDUMP}.t{self.runtime_config['cyc']:02d}z.letkfoi.yaml")
+
+        # Create a local dictionary that is repeatedly used across this class
+        local_dict = AttrDict(
+            {
+                'npx_ges': _res + 1,
+                'npy_ges': _res + 1,
+                'npz_ges': self.config.LEVS - 1,
+                'npz': self.config.LEVS - 1,
+                'npx_anl': _res_enkf + 1,
+                'npy_anl': _res_enkf + 1,
+                'npz_anl': self.config['LEVS'] - 1,
+                'LAND_WINDOW_BEGIN': _window_begin,
+                'LAND_WINDOW_LENGTH': f"PT{self.config['assim_freq']}H",
+                'OPREFIX': f"{self.runtime_config.CDUMP}.t{self.runtime_config.cyc:02d}z.",  # TODO: CDUMP is being replaced by RUN
+                'APREFIX': f"{self.runtime_config.CDUMP}.t{self.runtime_config.cyc:02d}z.",  # TODO: CDUMP is being replaced by RUN
+                'GPREFIX': f"gdas.t{self.runtime_config.previous_cycle.hour:02d}z.",
+                'fv3jedi_yaml': _fv3jedi_yaml,
+            }
+        )
+
+        # task_config is everything that this task should need
+        self.task_config = AttrDict(**self.config, **self.runtime_config, **local_dict)
+
+    @logit(logger)
+    def initialize(self: Analysis) -> None:
```

Comment:
Let @aerorahul confirm, but I think we may want to make each part of this a separate functions that all get called from here, e.g. `stage()`, a `generate_yaml()`, etc.